q_types.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python3
```

Comment:
I don't think we need a separate file for this code. It's probably good to keep all the actual code in one file for now